src/engine/x86-64/X86_64Interpreter.v3

```diff
@@ -2422,51 +2422,81 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 		}
 		bindHandler(Opcode.RESUME); {
 			var resumeStub = X86_64Label.new();
-			var cont_id = r_tmp0, r_ip_store = r_tmp1;
+			var cont_id = xenv.tmp0, ip_store = xenv.tmp1;
 			computeCurIpForTrap(-1);
 			computePcFromCurIp();
 			// read info and rewind %ip (need %ip at start of opcode during call)
-			asm.movq_r_r(r_ip_store, r_ip);
-			genReadUleb32(cont_id); // read signature index
-			asm.movq_r_r(r_ip, r_ip_store);
+			masm.emit_mov_r_r(ValueKind.REF, ip_store, xenv.ip);
+			genReadUleb32(G(cont_id)); // read signature index
+			masm.emit_mov_r_r(ValueKind.REF, xenv.ip, ip_store);
 			saveCallerIVars();
-			var r_cont = r_tmp1;
-			genResume(cont_id, r_cont, r_tmp3);
+
+			var cont = xenv.tmp1; // reuse due to non-overlapping live ranges
+			genPopInto(G(cont));
+			masm.emit_validate_and_consume_cont(xenv.tmp1);
+
 			// move parameter values from %curStack to %cont.top
-			var r_nvals = cont_id;
-			masm.emit_cont_mv(r_vsp, r_cont, r_nvals, r_tmp3, r_tmp4, r_xmm0);
+			var r_nvals = cont_id; // reuse due to non-overlapping live ranges
+			masm.emit_mov_r_m(ValueKind.REF, xenv.scratch, MasmAddr(xenv.instance, offsets.Instance_heaptypes));
```

Comment:
Following on above, we should have a `MacroAssembler` method for each field access; basically, `V3Offsets` should be used there.